aws/resource_aws_lb_listener_rule.go

```diff
@@ -920,3 +1234,115 @@ func highestListenerRulePriority(conn *elbv2.ELBV2, arn string) (priority int64,
 
 	return
 }
+
+func lbListenerRuleConditions(conditions []interface{}) ([]*elbv2.RuleCondition, error) {
+	elbConditions := make([]*elbv2.RuleCondition, len(conditions))
+	for i, condition := range conditions {
+		conditionMap := condition.(map[string]interface{})
+		conditionValues := conditionMap["values"].([]interface{})
+
+		elbConditions[i] = &elbv2.RuleCondition{
+			Field: aws.String(conditionMap["field"].(string)),
+		}
+		switch conditionMap["field"].(string) {
+		case "host-header":
+			hostHeader := conditionMap["host_header"].([]interface{})
+			var values []interface{}
+			if len(hostHeader) > 0 {
+				values = hostHeader[0].(map[string]interface{})["values"].([]interface{})
+			} else if len(conditionValues) > 0 {
+				// Backwards compatibility
+				values = conditionValues
+			} else {
+				return nil, errors.New("host_header must be set when condition field is host-header")
```

Comment:
To prevent handling errors at apply time for Create and Update, I'm wondering if we cannot handle such logic in CustomizeDiff, then deporting that earlier in the process of planning.
@bflad what do you think?

Reply:
I've done this. It's not something that is done much in the provider. There are numerous examples of resources doing apply time validation or just leaving it up to the AWS API.